Content.Server/GameObjects/Components/Body/Surgery/BiologicalSurgeryDataComponent.cs

```diff
@@ -94,94 +188,75 @@ public class BiologicalSurgeryDataComponent : SurgeryDataComponent
-                {
-                    toReturn += Loc.GetString("{0:their} {1} is loose.\n", Owner, mechanism.Name);
-                }
-            }
-
-            return toReturn;
-        }
-
-        public override bool CanAddMechanism(IMechanism mechanism)
-        {
-            return Parent != null &&
-                   _skinOpened &&
-                   _vesselsClamped &&
-                   _skinRetracted;
-        }
 
-        public override bool CanAttachBodyPart(IBodyPart part)
-        {
-            return Parent != null;
-            // TODO BODY if a part is disconnected, you should have to do some surgery to allow another bodypart to be attached.
+            step(container, surgeon, performer);
+            return true;
         }
 
-        private void OpenSkinSurgery(IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
+        private async void OpenSkinSurgery(IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
         {
-            if (Parent == null) return;
+            if (Parent == null)
+            {
+                return;
+            }
 
-                performer.PopupMessage(Loc.GetString("Cut open the skin..."));
+            performer.PopupMessage(Loc.GetString("Cut open the skin..."));
 
-            // TODO BODY do_after: Delay
-            _skinOpened = true;
+            if (await SurgeryDoAfter(performer))
+            {
+                SkinOpened = true;
+            }
         }
 
-        private void ClampVesselsSurgery(IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
+        private async void ClampVesselsSurgery(IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
         {
             if (Parent == null) return;
 
             performer.PopupMessage(Loc.GetString("Clamp the vessels..."));
 
-            // TODO BODY do_after: Delay
-            _vesselsClamped = true;
+            if (await SurgeryDoAfter(performer))
+            {
+                VesselsClamped = true;
+            }
         }
 
-        private void RetractSkinSurgery(IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
+        private async void RetractSkinSurgery(IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
         {
             if (Parent == null) return;
 
             performer.PopupMessage(Loc.GetString("Retract the skin..."));
 
-            // TODO BODY do_after: Delay
-            _skinRetracted = true;
+            if (await SurgeryDoAfter(performer))
+            {
+                SkinRetracted = true;
+            }
         }
 
-        private void CauterizeIncisionSurgery(IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
+        private async void CauterizeIncisionSurgery(IBodyPartContainer container, ISurgeon surgeon, IEntity performer)
         {
             if (Parent == null) return;
 
             performer.PopupMessage(Loc.GetString("Cauterize the incision..."));
```

Comment:
```suggestion
            performer.PopupMessage(Loc.GetString("Cauterizing the incision..."));
```

Ignore if surgery is getting refactored obviously.